common/src/com/thoughtworks/go/config/materials/git/GitMaterial.java

```diff
@@ -96,17 +115,18 @@ public GitMaterial(GitMaterialConfig config) {
 
     @Override
     public MaterialConfig config() {
-        return new GitMaterialConfig(url, branch, submoduleFolder, autoUpdate, filter, folder, name);
+        return new GitMaterialConfig(url, branch, submoduleFolder, autoUpdate, filter, folder, name, shallowClone);
     }
 
     public List<Modification> latestModification(File baseDir, final SubprocessExecutionContext execCtx) {
         ArrayList<Modification> mods = new ArrayList<Modification>();
-        mods.add(getGit(baseDir).latestModification());
+        mods.add(getGit(baseDir, DEFAULT_SHALLOW_CLONE_DEPTH).latestModification());
         return mods;
     }
 
     public List<Modification> modificationsSince(File baseDir, Revision revision, final SubprocessExecutionContext execCtx) {
-        GitCommand gitCommand = getGit(baseDir);
+        GitCommand gitCommand = getGit(baseDir, DEFAULT_SHALLOW_CLONE_DEPTH);
+        unshallowIfNeeded(gitCommand, ProcessOutputStreamConsumer.inMemoryConsumer(), revision, baseDir);
```

Comment:
@jyotisingh 

The main reason is to keep behavior consistency throughout GitMaterial implementation. The implementation should not assume how to it get used.  Make it consistent helps reduce surprises when usage context changed: e.g someday a developer need call modificationsSince somewhere else would wonder "why the hell I can not find a revision?"

Another (more practical) reason is to make it robust against external changes.  In cases of flyweight repo corrupted, user can just rm it and let go clone it again. Without `unshallowIfNeeded ` here, they will lost abilities of triggering old build.